cooked-validators/src/Cooked/Tweak/OutPermutations.hs

```diff
@@ -73,3 +79,28 @@ nonIdentityPermutations l = removeFirst l $ distinctPermutations l
     removeFirst :: Eq a => a -> [a] -> [a]
     removeFirst _ [] = []
     removeFirst x (y : ys) = if x == y then ys else y : removeFirst x ys
+
+-- * Random shuffle algorithm taken from https://wiki.haskell.org/Random_shuffle
+
+fisherYatesStep :: RandomGen g => (Map Int a, g) -> (Int, a) -> (Map Int a, g)
+fisherYatesStep (m, gen) (i, x) = ((Map.insert j x . Map.insert i (m Map.! j)) m, gen')
+  where
+    (j, gen') = randomR (0, i) gen
+
+fisherYates :: RandomGen g => g -> [a] -> ([a], g)
+fisherYates gen [] = ([], gen)
+fisherYates gen l =
+  toElems $ foldl fisherYatesStep (initial (head l) gen) (numerate (tail l))
+  where
+    toElems (x, y) = (Map.elems x, y)
+    numerate = zip [1 ..]
+    initial x = (Map.singleton 0 x,)
```

Comment:
Hmm, what about, say, [random-shuffle](https://hackage.haskell.org/package/random-shuffle-0.0.4/docs/System-Random-Shuffle.html)?

Reply:
I'd like to use it as it seems to be more standard.
Will I be able to get the same behavior, that is, provided seeds are differents, then permutations should be as well?

Reply:
They should be AFAIK (up to the usual statistical collisions).